File: Ser2023/Company/Company.Web/Modules/Company/AddressType/AddressTypeColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.AddressType")]
[BasedOnRow(typeof(AddressTypeRow), CheckNames = true)]
public class AddressTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long AddressTypeId { get; set; }
    [EditLink]
    public string AddressTypeName { get; set; }
    //public Guid Rowguid { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/AddressTypeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AddressTypeRow } from "./AddressTypeRow";

export interface AddressTypeColumns {
    AddressTypeId: Column<AddressTypeRow>;
    AddressTypeName: Column<AddressTypeRow>;
}

export class AddressTypeColumns extends ColumnsBase<AddressTypeRow> {
    static readonly columnsKey = 'Company.AddressType';
    static readonly Fields = fieldsProxy<AddressTypeColumns>();
}